### ISSUE TITLE
StatusesVacuum can be more clear and configurable

### ISSUE THREAD

**Issue by renatolond** [mastodon/mastodon#21307]
### Pitch

Right now, the StatusesVacuum, which cleans old statuses, cleans up any old status. When I opened #21210 I noticed a few things which I felt could be more configurable.

1. The way to detect old status should be more clear
   Right now, the way old statues are cleared is by checking their ID against Mastodon's snowflake ID. Is that a snowflake ID created by Mastodon when ingesting the status? Or is it the original status ID? Since that influences which posts will be removed, I think it should be more clear
2. There should be ways to "protect" certain statuses from being removed. Right now all of them are removed. But I feel that private messages should not be removed. Neither should be boosts or bookmarked ones.
3. It would be nice to have an idea of how many status are going to be removed by a given config and a status of how many statuses were removed by previous runs of the vacuum.
4. There should be clearer instructions regarding DB maintenance after this is run. I run a test instances for a few years now and I had 3.3GB of DB usage when this first run, after I ran it, it deleted a massive amount of statuses, but no disk space was recovered because of how postgres works. The disk space will only be recovered by running a VACUUM FULL, which is requires a full lock of the table while running https://wiki.postgresql.org/wiki/VACUUM_FULL
  This should not be needed after the first run, since after that the periodic removal should keep the table at the right size of the window, but again, not clear.

### Motivation

This will be used by a lot of admins out there without knowing that this might remove data that their users might want to save.

**Comment by hugogameiro** [comment]
I completely agree with point 2 and had already suggested an idea to make the filtering of status to remove less resource intensive on https://github.com/mastodon/mastodon/discussions/19260 but probably there are better ways. Just something that came to my mind when thinking about ways to do this but avoid creating more database indexes.
